### PR TITLE
Battle logic: fix Genie special ability

### PR DESCRIPTION
**Pull request by idshibanov**
Self-explanatory title; fixing chance and the effect of special ability. Important to use it before any damage is applied.

Correct chance is pulled from HEROES2W file, it's checking for Genies at 0x04806CC.

### CODE REVIEW

File: src/fheroes2/battle/battle_action.cpp [hunk 1]
```diff
@@ -480,6 +480,20 @@ Battle::TargetsInfo Battle::Arena::GetTargetsForDamage( Unit & attacker, Unit &
     // first target
     res.defender = &defender;
     res.damage = attacker.GetDamage( defender );
+
+    // Genie special attack (20%, genie likes number 2)
+    if ( attacker.GetID() == Monster::GENIE && Rand::Get( 1, 5 ) == 2 ) {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_action.cpp:485]: Is it 20%? Most of sources point to 9-10%.

Comment (idshibanov) [reply src/fheroes2/battle/battle_action.cpp:485]: I debugged this couple times yesterday, this is exactly how it's checked in Gold version: Rand(1,5) == 2.

File: src/fheroes2/battle/battle_troop.cpp [hunk 2]
```diff
@@ -52,15 +47,6 @@ void Battle::UpdateMonsterAttributes( const std::string & spec )
     if ( doc.LoadFile( spec.c_str() ) && NULL != ( xml_battle = doc.FirstChildElement( "battle" ) ) ) {
         const TiXmlElement * xml_element;
         int value;
-
```
Comment (ihhub) [inline src/fheroes2/battle/battle_troop.cpp:55]: XML support is for modification of monster's abilities. We might need it but as of now it is okay to remove this chunk of code.

File: src/fheroes2/battle/battle_action.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/battle/battle_action.cpp:485]: I think it is a bug in the original game. According to the manual from the game `10% halve enemy unit` so let's make **10%**

Comment (ihhub) [reply src/fheroes2/battle/battle_action.cpp:485]: What do you think?

File: src/fheroes2/battle/battle_troop.cpp (hunk 2 above)
Comment (idshibanov) [reply src/fheroes2/battle/battle_troop.cpp:55]: This is a wrong file for that variable anyway.

File: src/fheroes2/battle/battle_action.cpp (hunk 1 above)
Comment (idshibanov) [reply src/fheroes2/battle/battle_action.cpp:485]: Alright, let's make it 10%.

**Comment by ihhub** [comment]
Hi @idshibanov , thank you very much for your contribution to the project!

**Pull request merged**